MdePkg/Library/SoftFloatLib/SoftFloatLib.inf

```diff
@@ -11,11 +11,11 @@
 
 [Defines]
   INF_VERSION                    = 0x0001001B
-  BASE_NAME                      = ArmSoftFloatLib
-  FILE_GUID                      = a485f921-749e-41a0-9f91-62f09a38721c
+  BASE_NAME                      = SoftFloatLib
+  FILE_GUID                      = C32116FA-C639-406D-950F-FE9B691122A8
   MODULE_TYPE                    = BASE
   VERSION_STRING                 = 1.0
-  LIBRARY_CLASS                  = ArmSoftFloatLib
+  LIBRARY_CLASS                  = SoftFloatLib

```

Comment:
(cannot respond to the commit log so I'll randomly reply here)

cc @leiflindholm @kraxel @samimujawar 

It is kind of unfortunate that we go through all this pain only to accommodate 32-bit ARM, which has no future and therefore very little relevance to development.

What would we lose by just removing this library, and dropping support for things like HTTPS boot and secure boot from 32-bit ARM platforms entirely?

Reply:
I fully agree.

Reply:
MbedTLS does not appear to require softfloat, so we might just switch 32-bit ARM to that. We will lose HTTPS boot but secure boot should still work (which is arguably more important, as it involves OS visible interfaces)

Reply:
Yes this approach looks good to me.

Reply:
Looks good to me too.